### PR TITLE
Added option to set timer type when signal is used

### PR DESCRIPTION
**Pull request by rowleya**
Implementation of #79

### CODE REVIEW

File: pytest_timeout.py [hunk 1]
```diff
@@ -66,29 +74,41 @@ def pytest_addoption(parser):
...
-        "timeout(timeout, method=None, func_only=False): Set a timeout, timeout "
+        'timeout(timeout, method=None, func_only=False, signal_timer="ITIMER_REAL"): '
+        "Set a timeout, timeout "
         "method and func_only evaluation on just one test item.  The first "
         "argument, *timeout*, is the timeout in seconds while the keyword, "
         "*method*, takes the same values as the --timeout_method option. The "
         "*func_only* keyword, when set to True, defers the timeout evaluation "
         "to only the test function body, ignoring the time it takes when "
-        "evaluating any fixtures used in the test.",
+        "evaluating any fixtures used in the test."
```
Comment (flub) [inline pytest_timeout.py:102]: This is missing a trailing space, line concatenation will turn this into `test.The` currently afaik.

File: pytest_timeout.py [hunk 2]
```diff
@@ -127,11 +147,13 @@ def pytest_report_header(config):
     """Add timeout config to pytest header."""
     if config._env_timeout:
         return [
-            "timeout: %ss\ntimeout method: %s\ntimeout func_only: %s"
+            "timeout: %ss\ntimeout method: %s\ntimeout func_only: %s\n"
+            "timeout signal_timer: %s"
```
Comment (flub) [inline pytest_timeout.py:151]: This is starting to take up a lot of space in the header.  Maybe we should just make it a single line instead?  Means also we don't have to keep repeating `timeout`.
```suggestion
            "timeout: %ss   method: %s   func_only: %s   signal_timer: %s"
```

File: pytest_timeout.py (hunk 1 above)
Comment (rowleya) [reply pytest_timeout.py]: Fixed

**Comment by flub** [comment]
hey, sorry for the delay here.  i think the code is fine, but I'm still really unconvinced of adding this complexity for such a super weird edge case.  I already have some features here that i accepted reluctantly because I felt bad rejecting a perfectly well made PR by someone, but afterwards regretted adding those features.  And I really fear this is a similar case.  This all seems an elaborate workaround for an overloaded CI server so far.  And in all these years you're the first one to think this is required.

**Comment by rowleya** [comment]
Surprised you think it is a "weird edge case" but if you don't want it, don't take it; we can always keep it in our own repository and use it as needed.

**Comment by flub** [comment]
going to close this, there hasn't been more demand for this and my reasoning probably didn't change in the mean time.

**Pull request closed without merging**